pallet-fuso-foundation/src/lib.rs

```diff
@@ -60,21 +73,42 @@ pub mod pallet {
     }
 
     #[pallet::genesis_build]
-    impl<T: Config> GenesisBuild<T> for GenesisConfig<T>
-    where
-        IdentifierOf<T>: From<(u8, [u8; 32])>,
-        <T as frame_system::Config>::AccountId: Into<[u8; 32]>,
-    {
+    impl<T: Config> GenesisBuild<T> for GenesisConfig<T> {
         fn build(&self) {
-            for (account, balance) in &self.fund {
-                Foundation::<T>::insert(account, balance);
+            for (account, _balance) in &self.fund {
+                Foundation::<T>::insert(
+                    account,
+                    FoundationData {
+                        deny_durations: _balance.0,
+                        interval_durations: _balance.1,
+                        times: _balance.2,
+                        amount: _balance.3,
+                    },
+                );
             }
             for (account, balance) in &self.fund_total {
-                <pallet_balances::Pallet<T>>::reserve(&account, *balance);
+                let _r = <pallet_balances::Pallet<T>>::reserve(&account, *balance);
```

Comment:
1. don't ignore result, use `.unwrap()` so we can know since it is genesis config;
2. `reserve` won't work since those accounts are all empty; you may never know because you don't handle the result of `reserve`. use `mutate_account` instead;
3. keep the style consistent `pallet_balances::Pallet<T>::...`;